src/devices/Ads1115/Ads1115.cs

```diff
@@ -104,14 +220,255 @@ private void SetConfig()
+                _gpioController.ClosePin(_alrtRdyPin);
+                _gpioController = null;
+                _alrtRdyPin = -1;
+            }
+        }
+
+        /// <summary>
+        /// Write the two comparator registers
+        /// </summary>
+        /// <param name="loThreshold">High threshold value (unsigned short)</param>
+        /// <param name="hiThreshold">Low threshold value (unsigned short)</param>
+        private void WriteComparatorRegisters(short loThreshold, short hiThreshold)
+        {
+            Span<byte> writeBuff = stackalloc byte[3]
+            {
+                (byte)Register.ADC_CONFIG_REG_LO_THRESH, (byte)(loThreshold >> 8), (byte)(loThreshold & 0xFF)
+            };
+            _i2cDevice.Write(writeBuff);
+            writeBuff = stackalloc byte[3]
+            {
+                (byte)Register.ADC_CONFIG_REG_HI_THRESH, (byte)(hiThreshold >> 8), (byte)(hiThreshold & 0xFF)
+            };
+            _i2cDevice.Write(writeBuff);
         }
 
         /// <summary>
-        /// Read Raw Data
+        /// Enable conversion ready event.
+        /// The <see cref="AlertReadyAsserted"/> event fires each time a new value is available after this method is called.
+        /// </summary>
+        /// <param name="gpioController">Reference to the GPIO controller</param>
+        /// <param name="pin">Pin to which the ALRT/RDY Pin is connected</param>
+        /// <exception cref="InvalidOperationException">The conversion ready event is already set up</exception>
+        public void EnableConversionReady(GpioController gpioController, int pin)
+        {
+            if (gpioController == null)
+            {
+                throw new ArgumentNullException(nameof(gpioController));
+            }
+
+            if (_gpioController != null)
+            {
+                throw new InvalidOperationException("Event mode already set.");
+            }
+
+            try
+            {
+                // The ALRT/RDY Pin requires a pull-up resistor
+                gpioController.OpenPin(pin, PinMode.InputPullUp);
+
+                // Must be set to something other than disable
+                _comparatorQueue = ComparatorQueue.AssertAfterOne;
+                _deviceMode = DeviceMode.Continuous;
+                SetConfig();
+                // Writing a negative value to the max value register and a positive value to the min value register
+                // configures the ALRT/RDY pin to trigger after each conversion (with a transition from high to low, when ComparatorPolarity is Low)
+                WriteComparatorRegisters(short.MaxValue, short.MinValue);
+
+                gpioController.RegisterCallbackForPinValueChangedEvent(pin, ComparatorPolarity == ComparatorPolarity.Low ? PinEventTypes.Falling : PinEventTypes.Rising, ConversionReadyCallback);
+
+            }
+            catch (Exception)
+            {
+                gpioController.ClosePin(pin);
```

Comment:
you can put this in `finally` clause to achieve the same effect

Reply:
No. This catch handler is only executed if an error occurs. If everything succeeds, the pin must not be closed here.